Classes/Hook/FileDumpControllerHook.php

```diff
@@ -0,0 +1,177 @@
+
+            $applications = $this->getLinkedApplications($file->getUid());
+
+            $hasAccess = false;
+            foreach ($applications as $application) {
+                if ($this->hasAccess($application, $feUser, $beUser)) {
+                    $hasAccess = true;
+                    break;
+                }
+            }
+
+            if (!$hasAccess) {
+                HttpUtility::setResponseCodeAndExit(HttpUtility::HTTP_STATUS_403);
+            }
+        }
+    }
+
+    /**
+     * Returns true if the current file is located inside the protected ATS folder.
+     *
+     * @param  AbstractFile $file
+     * @return bool
+     */
+    public function fileIsInAtsStorage(AbstractFile $file)
+    {
+        return StringUtility::beginsWith($file->getCombinedIdentifier(), ExtconfService::getInstance()->getUploadConfiguration()['uploadFolder']);
+    }
+
+    /**
+     * Checks if the given be/fe user combination has access to the current file
+     *
+     * @todo currently all logged in backend users have access to protected files.
+     * This should be narrowed down further to check if they actually have permission to view the corresponding application.
+     * Since there are a lot of factors for this permission check (access to listAll-Action, extensions like pagemachine/extbase-acl),
+     * permission handling needs simplification/cleanup before it can be evaluated here.
+     *
+     * @param  array  $application
+     * @param  FrontendUserAuthentication $feUser
+     * @param  FrontendBackendUserAuthentication  $beUser
+     * @return bool
+     */
+    public function hasAccess($application, FrontendUserAuthentication $feUser = null, FrontendBackendUserAuthentication $beUser = null)
+    {
+        $granted = false;
+        if ($feUser->user !== null) {
+            if ($feUser->user['uid'] == $application['user']) {
+                $granted = true;
+            }
+        }
+
+        if ($beUser !== null) {
+            $granted = true;
+        }
+
+        return $granted;
+    }
+
+    /**
+     * Returns all applications connected to a file
+     *
+     * @param  int $fileUid
+     * @return array
+     */
+    protected function getLinkedApplications($fileUid)
+    {
+        $constraints = [
+            sprintf('sys_file_reference.tablenames = "%s"', $this->tableNames),
+            sprintf('sys_file_reference.fieldname = "%s"', $this->fieldName),
+            sprintf('sys_file.uid = "%s"', intval($fileUid)),
+        ];
+
+        $applicationQuery = $this->getDatabaseConnection()->exec_SELECT_mm_query(
+            'tx_ats_domain_model_application.*',
+            'sys_file',
+            'sys_file_reference',
+            'tx_ats_domain_model_application',
+            ' AND ' . implode(' AND ', $constraints)
+        );
+
+        $applications = [];
+
+        while ($row = $this->getDatabaseConnection()->sql_fetch_assoc($applicationQuery)) {
```

Comment:
The whole DB interaction can be simplified to `exec_SELECTgetRows()` here. Alternatively this method could use a generator instead.